src/liberty/devCommands.ts

```diff
@@ -7,528 +7,595 @@
-        (result!==0)? console.error(message):console.info(message);projectProvider.fireChangeEvent();
-        vscode.window.showInformationMessage(message);
-        
-    } else {
-        // clicked on the empty space and workspace has more than one folders, or
-        // from command palette
-        // Display the list of workspace folders for user to select.
-        // The list should not contain any existing projects
-        const uris: string[] = [];
-        const wsFolders = vscode.workspace.workspaceFolders;
-		if ( wsFolders ) {
-			for ( const folder of wsFolders ) {
-				const path = folder.uri.fsPath;
-				if ( projectProvider.projectRootPathExists(path, projectProvider.getProjects().keys() ) === false ) {
-					uris.push(folder.uri.fsPath);
-				}
-			}
-		}
-        if ( uris.length === 0 ) {
-            // show error
-            const message = localize("add.project.manually.no.projects.available.to.add");
-            console.error(message);
-            vscode.window.showInformationMessage(message);
-        } else {
-            // present the list
-            vscode.window.showQuickPick(uris).then(async selection => {
-                if (!selection) {
-                    return;
+        if ( paths.length === 1) {
+            console.log(localize("attach.debugger.liverty.dev.in",libProject.getLabel()));
+            const file = Path.resolve(paths[0]);
+            const lines = await fse.readFileSync(file, "utf8").split("\n");
+            let port = "";
+            for ( let i = 0; i < lines.length && port === ""; i++ ) {
+                const line = lines[i];
+                let match = undefined;
+                if ((match = LIBERTY_SERVER_ENV_PORT_REGEX.exec(line)) !== null) {
+                    port = match[1];
                 }
-                const result = await projectProvider.addUserSelectedPath (selection, projectProvider.getProjects());
-                const message = localize(`add.project.manually.message.${result}`, selection);
-                (result!==0)? console.error(message):console.info(message);projectProvider.fireChangeEvent();
-                vscode.window.showInformationMessage(message);
-            });
-        }
-    } 
-}
-// stop dev mode
-export async function stopDevMode(libProject?: LibertyProject | undefined): Promise<void> {
-    if (libProject !== undefined) {
-        console.log(localize("stopping.liverty.dev.on",libProject.getLabel()));
-        const terminal = libProject.getTerminal();
-        if (terminal !== undefined) {
-            terminal.show();
-            terminal.sendText("exit"); // stop dev mode on current project
+            }  
+           if (port !== "") {   
+               const path = Path.dirname(libProject.getPath()); 
+               const workspaceFolder = vscode.workspace.getWorkspaceFolder(vscode.Uri.file(libProject.getPath()));       
+               vscode.debug.startDebugging(workspaceFolder, {
+                   "type": "java",
```

Comment:
This looks to require the Java debugger. We should declare an explicit dependency in our package.json on the Java Debugger VS Code extension

Reply:
added